x-pack/legacy/plugins/apm/server/lib/helpers/es_client.ts

```diff
@@ -181,6 +185,24 @@ export function getESClient(
     },
     indicesCreate: (params: IndicesCreateParams) => {
       return withTime(() => callMethod('indices.create', params));
+    },
+    indicesPrivileges: (
+      indices: ApmIndicesConfig
+    ): Promise<IndexPrivileges> => {
+      return withTime(() =>
+        callMethod('transport.request', {
+          method: 'POST',
+          path: '/_security/user/_has_privileges',
+          body: {
+            index: [
+              indices['apm_oss.errorIndices'],
+              indices['apm_oss.metricsIndices'],
+              indices['apm_oss.transactionIndices'],
+              indices['apm_oss.spanIndices']
+            ].map(indice => ({ names: indice, privileges: ['read'] }))
+          }
+        })
+      );
```

Comment:
This is currently just a wrapper around the elasticsearch client. However, you are adding actual business-logic (specific indices to check permissions for). You can perhaps instead do:

```ts
hasPrivileges: (params): Promise<IndexPrivileges> => {
  return withTime(() =>
    callMethod("transport.request", {
      method: "POST",
      path: "/_security/user/_has_privileges",
      body: body
    })
  );
};
```